File: group_project/group_project/src/main/java/com/dikshanta/restaurant/management/system/group_project/model/entities/Review.java
package com.dikshanta.restaurant.management.system.group_project.model.entities;

import com.dikshanta.restaurant.management.system.group_project.model.DateAuditable;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "reviews")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Review extends DateAuditable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Integer rating; // 1-5
    private String comment;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "restaurant_id")
    private Restaurant restaurant;

    @ManyToOne
    @JoinColumn(name = "food_item_id")
    private FoodItem foodItem;
}

File: group_project/frontend/src/types/review.ts
export interface ReviewRequest {
  rating: number;
  comment?: string;
  restaurantId?: number;
  foodItemId?: number;
}

export interface ReviewResponse {
  id: number;
  rating: number;
  comment?: string;
  userId: number;
  userName: string;
  restaurantId?: number;
  foodItemId?: number;
  createdAt: string;
}
